cmd/juju/cloud/add_test.go

```diff
@@ -187,240 +271,225 @@ func (s *addSuite) TestInteractive(c *gc.C) {
 	err = command.Run(ctx)
 	c.Check(errors.Cause(err), gc.Equals, io.EOF)
 
-	c.Assert(out.String(), gc.Equals, `
-Cloud Types
-  maas
-  manual
-  openstack
-  vsphere
-
-Select cloud type: 
-`[1:])
+	c.Assert(out.String(), gc.Equals, ""+
+		"Cloud Types\n"+
+		"  maas\n"+
+		"  manual\n"+
+		"  openstack\n"+
+		"  vsphere\n"+
+		"\n"+
+		"Select cloud type: \n",
+	)
 }
 
 func (s *addSuite) TestInteractiveOpenstack(c *gc.C) {
-	s.createTestCloudData(c)
-	command := cloud.NewAddCloudCommand()
+	fake := newFakeCloudQueryer()
+	fake.Call("PublicCloudMetadata", []string(nil)).Returns(map[string]cloudfile.Cloud{}, false, nil)
+	fake.Call("PersonalCloudMetadata").Returns(map[string]cloudfile.Cloud{}, nil)
+	myOpenstack := cloudfile.Cloud{
+		Type:      "openstack",
+		AuthTypes: []cloudfile.AuthType{"userpass", "access-key"},
+		Endpoint:  "http://myopenstack",
+		Regions: []cloudfile.Region{
+			{
+				Name:     "regionone",
+				Endpoint: "http://boston/1.0",
+			},
+		},
+	}
+	fake.Call("ParseOneCloud", []byte("auth-types:\n- userpass\n- access-key\nendpoint: http://myopenstack\nregions:\n  regionone:\n    endpoint: http://boston/1.0\n")).Returns(myOpenstack, nil)
```

Comment:
I think this would be a lot easier to read as a multiline string of some sort.  Having a one line version makes it very hard to know if it's a reasonable input or not, and if the format changes, it would be really hard to edit to make correct (like if we decide we want the indent to be 3 spaces instead of 2).